IHP/Job/Dashboard.hs

```diff
@@ -0,0 +1,530 @@
+    makeSection :: (?context :: ControllerContext, ?modelContext :: ModelContext) => IO SomeView
+
+    -- | The content of the page that will be displayed for a detail view of this job.
+    -- By default, the ID, Status, Created/Updated at times, and last error are displayed.
+    -- Can be defined as any arbitrary view.
+    makeDetailView :: (?context :: ControllerContext, ?modelContext :: ModelContext) => job -> IO SomeView
+    makeDetailView job = do
+        pure $ SomeView $ GenericShowView job
+
+    -- | The content of the page that will be displayed for the "new job" form of this job.
+    -- By default, only the submit button is rendered. For additonal form data, define your own implementation.
+    -- See 'GenericNewJobView' for a guide on how it can look.look
+    -- Can be defined as any arbitrary view, but it should be a form.
+    makeNewJobView :: (?context :: ControllerContext, ?modelContext :: ModelContext) => IO SomeView
+    makeNewJobView = pure $ SomeView $ GenericNewJobView @job
+
+    -- | The action run to create and insert a new value of this job into the database.
+    -- By default, create an empty record and insert it.
+    -- To add more data, define your own implementation.
+    createNewJob :: (?context :: ControllerContext, ?modelContext :: ModelContext) => IO ()
+    createNewJob = do
+        newRecord @job |> create
+        pure ()
+
+
+
+data TableView a = TableView [a]
+
+-- A TableView is only a View if it's contents are TableViewable.
+instance forall a. (TableViewable a) => View (TableView a) where
+    html (TableView rows) = renderTable rows
+
+-- | Creates a form with a single small button that links to the Create Job form for some 'DisplayableJob;.
+newJobFormForTableHeader :: forall job. (DisplayableJob job) => Html
+newJobFormForTableHeader =
+        let
+            t = tableName @job
+            link :: Text = "/jobs/CreateJob?tableName=" <> t
+        in [hsx|
+            <form action={link}>
+                <button type="submit" class="btn btn-primary btn-sm">+ New Job</button>
+            </form>
+        |]
+
+-- | Defines a set of necessary functions to display some data 'a' in a table format given a list of '[a]'.
+class TableViewable a where
+    tableTitle :: Text
+
+    -- | Headers that describe each column in the table
+    tableHeaders :: [Text]
+
+    -- | Optional HTML that will be placed in the top right of the table. Used for a "create new" button.
+    createNewForm :: Html
+    createNewForm = [hsx||]
+
+    renderTableRow :: a -> Html
+
+    renderTable :: [a] -> Html
+    renderTable rows =
+        let
+            title = tableTitle @a
+            headers = tableHeaders @a
+            renderRow = renderTableRow @a
+            newForm = createNewForm @a
+        in [hsx|
+        <div>
+            <div class="d-flex justify-content-between align-items-center">
+                <h3>Job type: {title}</h3>
```

Comment:
To simplify the interface we could remove the `Job type:` here (the `title` already has `jobs` in it, so this is kind of redundant)

What do you think about applying a table-name-to-human-readable-name transformation (e.g. like the forms is doing here https://github.com/digitallyinduced/ihp/blob/master/IHP/View/Form.hs#L738)?

So e.g. 

![image](https://user-images.githubusercontent.com/2072185/112449760-0aa09f00-8d54-11eb-8d66-f195003aff79.png)


would become this:

![image](https://user-images.githubusercontent.com/2072185/112449797-1724f780-8d54-11eb-997c-2614f770cbbe.png)